packages/block-library/src/image/edit.js

```diff
@@ -470,45 +623,57 @@ class ImageEdit extends Component {
 									type="number"
 									className="block-library-image__dimensions__width"
 									label={ __( 'Width' ) }
-									value={ width !== undefined ? width : '' }
+									value={ userSet ? width : '' }
 									placeholder={ imageWidth }
 									min={ 1 }
-									onChange={ this.updateWidth }
+									onChange={ ( value ) => {
+										this.updateWidth( value, imageWidth, imageHeight, true );
+									} }
 								/>
 								<TextControl
 									type="number"
 									className="block-library-image__dimensions__height"
 									label={ __( 'Height' ) }
-									value={ height !== undefined ? height : '' }
+									value={ userSet ? height : '' }
 									placeholder={ imageHeight }
 									min={ 1 }
-									onChange={ this.updateHeight }
+									onChange={ ( value ) => {
+										this.updateHeight( value, imageWidth, imageHeight, true );
+									} }
 								/>
 							</div>
 							<div className="block-library-image__dimensions__row">
 								<ButtonGroup aria-label={ __( 'Image Size' ) }>
-									{ [ 25, 50, 75, 100 ].map( ( scale ) => {
-										const scaledWidth = Math.round( imageWidth * ( scale / 100 ) );
-										const scaledHeight = Math.round( imageHeight * ( scale / 100 ) );
-
-										const isCurrent = width === scaledWidth && height === scaledHeight;
+									{ [ 25, 50, 75, 100 ].map( ( percent ) => {
+										const blockWidth = getBlockWidth();
```

Comment:
There's no way the block could know its own width on an initial render.

Reply:
`getBlockWidth()` doesn't necessarily get this specific block's width. It rather gets the "visual" editor width for all blocks. This is needed as Gutenberg allows themes to set the editor width from CSS, and some (TwentyNineteen) do "fluid" width.